01-import_and_filter.py

```diff
@@ -9,90 +9,90 @@
-                 (raw_fname_in, subject))
-            continue
-
-        raw = mne.io.read_raw_fif(raw_fname_in,
-                                  allow_maxshield=config.allow_maxshield,
-                                  preload=True, verbose='error')
-
-        # add bad channels
-        raw.info['bads'] = bads
-        print("added bads: ", raw.info['bads'])
-
-        if config.set_channel_types is not None:
-            raw.set_channel_types(config.set_channel_types)
-        if config.rename_channels is not None:
-            raw.rename_channels(config.rename_channels)
-
-        # Band-pass the data channels (MEG and EEG)
-        print("Filtering data between %s and %s (Hz)" %
-              (config.l_freq, config.h_freq))
-        raw.filter(
-            config.l_freq, config.h_freq,
-            l_trans_bandwidth=config.l_trans_bandwidth,
-            h_trans_bandwidth=config.h_trans_bandwidth,
-            filter_length='auto', phase='zero', fir_window='hamming',
-            fir_design='firwin')
-
-        if config.resample_sfreq:
-            print("Resampling data to %.1f Hz" % config.resample_sfreq)
-            raw.resample(config.resample_sfreq, npad='auto')
-
-        raw.save(raw_fname_out, overwrite=True)
-        n_raws += 1
-
-        if config.plot:
-            # plot raw data
-            raw.plot(n_channels=50, butterfly=True, group_by='position')
-
-            # plot power spectral densitiy
-            raw.plot_psd(area_mode='range', tmin=10.0, tmax=100.0,
-                         fmin=0., fmax=50., average=True)
-
-    if n_raws == 0:
-        raise ValueError('No input raw data found.')
+    """Filter data from a single subject."""
+    print('Processing subject: {}'.format(subject))
+
+    # Construct the search path for the data file
+    subject_path = op.join(subject, config.kind, config.ses)
+    data_dir = op.join(config.bids_root, subject_path)
+
+    bids_basename = make_bids_basename(subject=subject,
+                                       session=config.ses,
+                                       task=config.task,
+                                       acquisition=config.acq,
+                                       run=config.run,
+                                       processing=config.proc,
+                                       recording=config.rec,
+                                       space=config.space
+                                       )
+
+    # Find the data file
+    search_str = op.join(data_dir, bids_basename) + '*'
+    fname_candidates = glob.glob(search_str)
+
+    if len(fname_candidates) == 1:
+        bids_fname = fname_candidates[0]
+    elif len(fname_candidates) == 0:
+        raise ValueError('Could not find input data file: "{}"'.format)
+    elif len(fname_candidates) > 1:
+        raise ValueError('Expected to find a single input data file: "{}" but '
+                         'found:\n\n{}'.format(search_str, fname_candidates))
```

Comment:
This should also happen inside `mne-bids`? Basically I would give the filename without the extension.